plugins/repository-s3/build.gradle

```diff
@@ -169,177 +147,39 @@ buildscript {
-  final String minioFilePath = "${gradle.gradleUserHomeDir}/downloads/minio/${minioDistribution}/${minioFileName}"
+if (useFixture) {
 
-  task downloadMinio(type: Download) {
-    src minioDownloadURL
-    dest minioFilePath
-    onlyIfModified true
-  }
+  apply plugin: 'elasticsearch.test.fixtures'
 
-  task verifyMinioChecksum(type: Verify, dependsOn: downloadMinio) {
-    src minioFilePath
-    algorithm 'SHA-256'
-    checksum minioCheckSum
-  }
-
-  task installMinio(type: Sync, dependsOn: verifyMinioChecksum) {
-    from minioFilePath
-    into minioBinDir
-    fileMode 0755
+  RestIntegTestTask integTestMinio = project.tasks.create('integTestMinio', RestIntegTestTask.class) {
+    description = "Runs REST tests using the Minio repository."
   }
 
-  task startMinio {
-    dependsOn installMinio
-
-    ext.minioPid = 0L
-    ext.minioPort = 0
-
+  Task writeDockerFile = project.tasks.create('writeDockerFile') {
+    File minioDockerfile = new File("${project.buildDir}/minio-docker/Dockerfile")
+    outputs.file(minioDockerfile)
     doLast {
-      new File("${minioDataDir}/${s3PermanentBucket}").mkdirs()
-
-      Exception accumulatedException = null
-      for (int i = 0; i < 5; ++i) {
-        try {
-          minioPort = freePort(minioAddress)
-          final Process process =
-            setupMinio(minioAddress, minioPort, minioDataDir, s3PermanentAccessKey, s3PermanentSecretKey, minioBinDir, minioFileName)
-          minioPid = getMinioPid(process)
-          addShutdownHook(process, minioPort, minioPid)
-          break
-        } catch (Exception e) {
-          logger.error("Exception while trying to start Minio {}", e)
-          if (accumulatedException == null) {
-            accumulated = e
-          } else {
-            accumulatedException.addSuppressed(e)
-          }
-        }
-      }
-      if (accumulatedException != null) {
-        throw new GradleException("Failed to start Minio", accumulatedException)
-      }
+      minioDockerfile.parentFile.mkdirs()
+      minioDockerfile.text = "FROM minio/minio:RELEASE.2019-01-23T23-18-58Z\n" +
+        "RUN mkdir -p /minio/data/${s3PermanentBucket}\n" +
+        "ENV MINIO_ACCESS_KEY ${s3PermanentAccessKey}\n" +
+        "ENV MINIO_SECRET_KEY ${s3PermanentSecretKey}"
     }
   }
 
-  task stopMinio(type: LoggedExec) {
-    onlyIf { startMinio.minioPid > 0 }
-
-    doFirst {
-      logger.info("Shutting down minio with pid ${startMinio.minioPid}")
-    }
-
-    final Object pid = "${ -> startMinio.minioPid }"
-
-    // we skip these tests on Windows so we do no need to worry about compatibility here
-    executable = 'kill'
-    args('-9', pid)
-  }
-
-  RestIntegTestTask integTestMinio = project.tasks.create('integTestMinio', RestIntegTestTask.class) {
-    description = "Runs REST tests using the Minio repository."
-  }
-
+  composeUp.dependsOn(writeDockerFile)
```

Comment:
I wouldn't want to use these tasks directly. We have a `buildFixture` task. 
You could add a `doLast` to `preProcessFixture` to write the docker file, or have `preProcessFixture`  depends on `writeDockerFile`

Reply:
done :)